Adding Shield-Advanced files to support upcoming TDS blog

Jsmiiii opened the pull request:
Adding:

- drt-removal Lambda Node.js function. This allow DRT permissions
to be revoked from S3 and Shield resources, once a DDoS incident
is confirmed as closed.
- shield-engagement Lambda Node.js function. This automatically
launches a support ticket, routes a message to the DRT SNS topic and
grants DRT for access to Shield and S3 resources, to aid them with
analysis and diag tasks during a DDoS incident.
- add-protections.sh Bash script which automatically adds new tagged
AWS resources as shield protected resources.

Full info/instructions in the individual Readme's.

WillJCJ commented on aws-lambda-drt-removal/.eslintignore:
```diff
@@ -0,0 +1,5 @@
+.circleci
```
I assume you don't need `.circleci` and `terraform` seeing as you aren't including them in the repo?

Jsmiiii replied:
No, not required. I've now removed these.

WillJCJ commented on aws-lambda-shield-engagement/README.md:
```diff
@@ -0,0 +1,120 @@
+
+You should create an IAM role which the Lambda uses as an execution role. This role should have a trusted relationship to AWS Lambda as per:
+
+{
```
This could be surrounded in ``` with JSON syntax highlighting so that Markdown doesn't squish it into one line.

Jsmiiii replied:
Thanks and now corrected!

Jsmiiii committed:
Adding Shield-Advanced files to support upcoming TDS blog

Adding:

- drt-removal Lambda Node.js function. This allow DRT permissions
to be revoked from S3 and Shield resources, once a DDoS incident
is confirmed as closed.
- shield-engagement Lambda Node.js function. This automatically
launches a support ticket, routes a message to the DRT SNS topic and
grants DRT for access to Shield and S3 resources, to aid them with
analysis and diag tasks during a DDoS incident.
- add-protections.sh Bash script which automatically adds new tagged
AWS resources as shield protected resources.

Full info/instructions in the individual Readme's.

Jsmiiii committed:
Adding Apache Open Source License terms.

PsypherPunk reviewed (approved):
LGTM.
